test/common/buffer/owned_impl_test.cc

```diff
@@ -76,6 +76,54 @@ TEST_F(OwnedImplTest, AddBufferFragmentDynamicAllocation) {
   EXPECT_TRUE(release_callback_called_);
 }
 
+TEST_F(OwnedImplTest, Prepend) {
```

Comment:
I think this is now testing the same function as PrependString yeah?  If so let's remove.

Reply:
Removed redundant tests from `OwnedImplTest` & `WatermarkBufferTest`